hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/AdjustAllowanceChecks.java

```diff
@@ -0,0 +1,290 @@
+	}
+
+	@Override
+	public ResponseCodeEnum validateCryptoAllowances(final List<CryptoAllowance> cryptoAllowancesList,
+			final Account ownerAccount) {
+		if (cryptoAllowancesList.isEmpty()) {
+			return OK;
+		}
+
+		if (hasRepeatedSpender(cryptoAllowancesList.stream().map(CryptoAllowance::getSpender).toList())) {
+			return SPENDER_ACCOUNT_REPEATED_IN_ALLOWANCES;
+		}
+
+		final var existingAllowances = ownerAccount.getCryptoAllowances();
+		for (final var allowance : cryptoAllowancesList) {
+			final var spender = Id.fromGrpcAccount(allowance.getSpender());
+			final var allowanceOwner = Id.fromGrpcAccount(allowance.getOwner());
+			final var amount = allowance.getAmount();
+
+			final var key = spender.asEntityNum();
+			final var existingAmount = existingAllowances.containsKey(key) ? existingAllowances.get(key) : 0;
+
+			// validate if the total allowance combined with existing allowance < 0
+			var validity = validateAmount(amount, existingAmount);
+			if (validity != OK) {
+				return validity;
+			}
+			validity = validateCryptoAllowanceBasics(ownerAccount.getId(), allowanceOwner, spender);
+			if (validity != OK) {
+				return validity;
+			}
+		}
+		return OK;
+	}
+
+	@Override
+	public ResponseCodeEnum validateFungibleTokenAllowances(final List<TokenAllowance> tokenAllowancesList,
+			final Account ownerAccount) {
+		if (tokenAllowancesList.isEmpty()) {
+			return OK;
+		}
+
+		if (hasRepeatedId(
+				tokenAllowancesList.stream().map(a -> FcTokenAllowanceId.from(EntityNum.fromTokenId(a.getTokenId()),
+						EntityNum.fromAccountId(a.getSpender()))).toList())) {
+			return SPENDER_ACCOUNT_REPEATED_IN_ALLOWANCES;
+		}
+
+		final var existingAllowances = ownerAccount.getFungibleTokenAllowances();
+
+		for (final var allowance : tokenAllowancesList) {
+			final var spenderAccountId = allowance.getSpender();
+			final var owner = Id.fromGrpcAccount(allowance.getOwner());
+			final var amount = allowance.getAmount();
+			final var tokenId = allowance.getTokenId();
+			final var token = tokenStore.loadPossiblyPausedToken(Id.fromGrpcToken(tokenId));
+			final var spenderId = Id.fromGrpcAccount(spenderAccountId);
+
+			final var key = FcTokenAllowanceId.from(token.getId().asEntityNum(), spenderId.asEntityNum());
+			final var existingAllowance = existingAllowances.containsKey(key) ? existingAllowances.get(key) : 0;
+
+			if (!token.isFungibleCommon()) {
+				return NFT_IN_FUNGIBLE_TOKEN_ALLOWANCES;
+			}
+
+			var validity = validateTokenAmount(amount, existingAllowance, token);
+			if (validity != OK) {
+				return validity;
+			}
+
+			validity = validateTokenBasics(ownerAccount, spenderId, tokenId, owner);
+			if (validity != OK) {
+				return validity;
+			}
+		}
+		return OK;
+	}
+
+	@Override
+	public ResponseCodeEnum validateNftAllowances(final List<NftAllowance> nftAllowancesList,
+			final Account ownerAccount) {
+		if (nftAllowancesList.isEmpty()) {
+			return OK;
+		}
+
+		if (hasRepeatedId(
```

Comment:
This expression is hard to read inside an `if` condition. 😁

Reply:
Agreed 😄  . Moved to a local final variable and used in the condition